src/Rings/mpoly-ideals.jl

```diff
@@ -898,19 +898,24 @@ Multivariate polynomial ring in 2 variables over number field graded by
   y -> [1]
 ```
 """
-@attr function absolute_primary_decomposition(I::MPolyIdeal{<:MPolyRingElem{QQFieldElem}})
+@attr Vector{Tuple{typeof(I), typeof(I), ideal_type(change_base_ring(rationals_as_number_field()[1], base_ring(I))[1]), Int64}} function absolute_primary_decomposition(I::MPolyIdeal{<:MPolyRingElem{QQFieldElem}})
```

Comment:
I gave my best to get this function type stable. If anyone knows a better way to infer the return type, please speak up.

Reply:
Since the RK can either be `AbstractAlgebra.Generic.MPolyRing{AbsSimpleNumFieldElem}` or `MPolyDecRing{AbsSimpleNumFieldElem, AbstractAlgebra.Generic.MPolyRing{AbsSimpleNumFieldElem}}`, depending on if `base_ring(I)` is graded, this seems to be the straightforward way of resolving this. The only other idea in my head would be adding an extra dispatch for `I::MPolyIdeal{<:MPolyDecRing{QQFieldElem}}`, but this would imply a lot of code duplication